sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -596,4 +598,138 @@ public void countAssertsMultipleCallsIndependent() {
 
     assertThat(PAssert.countAsserts(pipeline), equalTo(3));
   }
+
+  @Test
+  public void testPAssertThatFlattened() {
```

Comment:
I think this needs `@Category(NeedsRunner.class)` which will cause it to be executed on the direct runner in the `:runners:direct-java:needsRunnerTests` task. See here: https://scans.gradle.com/s/rlhjzeofa5ytc/tests?toggled=WyI6cnVubmVyczpkaXJlY3QtamF2YS1uZWVkc3J1bm5lcnRlc3RzLW9yZy5hcGFjaGUuYmVhbS5zZGsuaW8iLCI6cnVubmVyczpkaXJlY3QtamF2YS1uZWVkc3J1bm5lcnRlc3RzLW9yZy5hcGFjaGUuYmVhbS5zZGsudHJhbnNmb3JtcyIsIjpydW5uZXJzOmRpcmVjdC1qYXZhLW5lZWRzcnVubmVydGVzdHMtb3JnLmFwYWNoZS5iZWFtLnNkay5zY2hlbWFzLnRyYW5zZm9ybXMiXQ#:runners:direct-java-needsrunnertests-org.apache.beam.sdk.testing-passerttest

This test was executed in 66ms according to Jenkins but in fact the `:sdks:java:core:test` task was pulled FROM-CACHE as I guess a prior run on this PR populated the cache: https://scans.gradle.com/s/rlhjzeofa5ytc/timeline#cblz6an6jai7a

I am actually surprised it did not fail since it should have abandoned node enforcement, which is a feature of the `@Rule TestPipeline` to protect users from writing tests but forgetting to run the pipeline.

Reply:
Thank you for the explanation! 
I fixed it.

Reply:
I see you added `ValidatesRunner` not `NeedsRunner`. This is actually OK. PAssert is not a primitive transform, so it does not fit the category. But a few runners do treat it special so it is good to include.